lib/Benchmark/Runner.php

```diff
@@ -125,6 +122,47 @@ private function runSubject(Benchmark $benchmark, Subject $subject)
         return $subjectResult;
     }
 
+    private function runIterations(Benchmark $benchmark, Subject $subject, $parameters)
+    {
+        $iterationResults = array();
+        for ($index = 0; $index < $subject->getNbIterations(); $index++) {
+            $iteration = new Iteration($index, $parameters);
+            $iterationResults[] = $this->runIteration($benchmark, $subject, $iteration);
+        }
+        return new IterationsResult($iterationResults, $parameters);
+    }
+
+    private function runIterationsSeparateProcess(Benchmark $benchmark, Subject $subject, $parameters)
+    {
+        $reflection = new \ReflectionClass(get_class($benchmark));
+
+        $bin = realpath(__DIR__ . '/../..') . '/bin/phpbench';
+        $command = sprintf(
+            '%s run %s --subject=%s --nosetup --dump --parameters=\'%s\'',
+            $bin,
+            $reflection->getFileName(),
+            $subject->getMethodName(),
+            json_encode($parameters)
+        );
+        exec($command, $output, $exitCode);
+        $output = implode(PHP_EOL, $output);
+
+        if (0 !== $exitCode) {
+            throw new \RuntimeException(sprintf(
+                'Failed to execute: "%s": %s',
+                $command,
+                $output
+            ));
+        }
+
+        $loader = new XmlLoader();
+        $suiteResult = $loader->load($output);
+        $iterationsResults = $suiteResult->getIterationsResults();
+        $iterationsResult = reset($iterationsResults);
```

Comment:
Validate that the number of iteration results is the same as the requested number.